.github/workflows/build.yml

```
@@ -0,0 +1,22 @@
+name: Basic Build
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '11', '17' ]
+    name: Basic Build (Java ${{ matrix.Java }})
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Setup java
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'adopt'
+        java-version: ${{ matrix.java }}
+    - name: Test
+      run: ./gradlew ciBasicBuild
```

There is a gradle-build-action action that will automatically cache various gradle build files between subsequent actions.

```suggestion
    - name: Test
       uses: gradle/gradle-build-action@v2
       with:
         arguments: ciBasicBuild
```

---

relevant action repo for reference: https://github.com/gradle/gradle-build-action

---

Interesting, but for now I will keep things simple and avoid caches. For now I think we can have a single job and step and just look at logs if something fails